doc/design/model_zoo.md

```diff
@@ -0,0 +1,174 @@
+1. Docker image ID, the Docker commit ID of the image `a_data_scientist/regressors` in the above example.
+1. submitter program, the source code of the submitter program, `my_first_model.py` in the above example, or its MD5 hash.
+1. data converter, the COLUMN and LABEL clauses.
+1. model parameter file path, the path to the trained model parameters on the distributed filesystem of the cluster.
+
+It is necessary to have the model ID so users can refer to the trained model when they want to use it.  Suppose that the user typed the prediction SQL statement at the beginning of this document. SQLFlow server will convert it into a submitter program and run it with the Docker image used to train the model. Therefore, the Docker image ID is also required. The model parameter path allows the prediction submitter program to locate and load the trained models.  The data converter helps the prediction submitter to use the conversion rules consistent with the ones used when training.
+
+It is necessary to record the content or the MD5 hash of the training submitter program in the model zoo table for experiment management. Please be aware that the training submitter encodes all three categories of hyperparameters, as listed in the above sections.  Suppose that the analyst re-trains the model with different hyperparameter settings, the training submitter changes accordingly, and SQLFlow should be able to remind the analyst to either uses a new model ID or overwrites the existing row in the model zoo table.
+
+### Model Sharing
+
+After all, what is a model zoo? A model zoo refers to all the model definitions and trained models accessible by a deployment of SQLFlow.  It contains one or more model definition Docker images and the source code repositories that build the images.  It also includes the model zoo table configured to work with all submitter programs generated by the deployment.
+
+Within a deployment, it is straightforward to share a trained model.  If the analyst, `an_analyst`, in the above example wants to use her own trained model `my_first_model` for prediction, she could use the short name `my_first_model`.
+
+```sql
+SELECT ... TO PREDICT ... USING my_first_model
+```
+
+If another analyst wants to use the trained model, he would need to use the full name as `an_analyst/my_first_model`.
+
+```sql
+SELECT ... TO PREDICT ... USING an_analyst/my_first_model
+```
+
+There could be more than one model definitions in each model's Docker image. We need to be able to find out which model definition is used to train current saved model. Also if we only want to reuse the model definition to train a new model we need to know the model definition class name, so that we can pass it to the `TRAIN` clause.
+
+To list all trained models of one user, you can do:
+
+```sql
+SQLFLOW LIST an_analyst
+```
+
+This should output a table showing the saved models and which model definition it was using.
```

Comment:
Is this table the same as the model zoo table mentioned above in line #86? It seems they have different schemas.

Reply:
I don't see relation to #86.